### PR TITLE
`polkadot-parachain`: Unify asset-hub authoring codepath with general AURA

### PR DESCRIPTION
**Pull request by skunert**
Recently thought about the special handling we have for asset-hub chains. They started with relay chain consensus and transitioned to AURA at some point. However, nobody should be authoring with relay chain consensus on these chains anymore, the transition is long done.

I propose to remove this special handling, allowing us to unify one more execution path.

### CODE REVIEW

File: cumulus/polkadot-parachain/src/service.rs [hunk 1]
```diff
@@ -635,109 +630,42 @@ where
 		para_id,
 		build_parachain_rpc_extensions::<RuntimeApi>,
```
Comment (bkchr) [inline cumulus/polkadot-parachain/src/service.rs:631]: This entire function is not required.

**Review by bkchr — COMMENTED** [review]
If we do this, we should do it properly. Basically all the `start_whatever_node` can just be deleted and all of them can just use `start_basic_lookahead_node`.

**Review by davxy — APPROVED** [review]

**Comment by serban300** [comment]
> If we do this, we should do it properly. Basically all the `start_whatever_node` can just be deleted and all of them can just use `start_basic_lookahead_node`.

If there are no concerns related to the conceptual change in this PR, I think it would be ok to merge it as it is.

I have a lot of cleanup related to `start_whatever_node` prepared in https://github.com/paritytech/polkadot-sdk/pull/4916 and after this PR is merged, I can adapt #4916 accordingly

**Review by serban300 — COMMENTED** [review]
I'm not very familiar with the consensus logic. I would prefer if someone with more knowledge of this logic would take a look also. But apart from that looks good to me.

**Review by bkchr — APPROVED** [review]
If @serban300 is getting rid of this in his next pr, we can continue.

**Pull request merged**